posydon/utils/common_functions.py

```diff
@@ -222,7 +223,7 @@ def roche_lobe_radius(m1, m2, a_orb=1):
         ## if array is empty, fill with NaN values
         if a_orb.size == 0:
             Pwarn("Trying to compute RL radius for binary with invalid separation", "EvolutionWarning")
-            a_orb = np.full_like(a_orb, np.nan, dtype=np.float64)
+            a_orb = np.full_like(a_orb.shape, np.nan, dtype=np.float64)
```

Comment:
This won't work:
`np.full_like` needs a array like object as first argument, while `np.full` takes a shape as first argument.

Reply:
The code I have written does work, I have tested it, and the code you have suggested does not work. Please see a screenshot below as to why if you are struggling to understand:

<img width="398" alt="Screen Shot 2024-09-07 at 8 40 08 PM" src="https://github.com/user-attachments/assets/2760604a-a3c8-4588-a091-dbf4643f7826">

Reply:
This does work for certain arrays (namely np.array([]) objects) because shape itself is array-like. I realized it does not work for arrays with more dimensions, please see most recent changes for that.